pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -151,5 +177,732 @@ func CreateCustomResourceDefinitions(clientset apiextensionsclient.Interface) er
 		return err
 	}
 
+	log.Info("Creating v2.CiliumNetworkPolicy CustomResourceDefinition... DONE")
+
 	return nil
 }
+
+func getStr(str string) *string {
+	return &str
+}
+
+func getInt64(i int64) *int64 {
+	return &i
+}
+
+var (
+	crv = apiextensionsv1beta1.CustomResourceValidation{
+		OpenAPIV3Schema: &apiextensionsv1beta1.JSONSchemaProps{
+			OneOf: []apiextensionsv1beta1.JSONSchemaProps{
+				{
+					Required: []string{"spec"},
+				},
+				{
+					Required: []string{"specs"},
+				},
+			},
+			Properties: properties,
+		},
+	}
+
+	properties = map[string]apiextensionsv1beta1.JSONSchemaProps{
+		"CIDR": {
+			Description: "CIDR is a CIDR prefix / IP Block.",
+			Type:        "string",
+			OneOf: []apiextensionsv1beta1.JSONSchemaProps{
+				{
+					// IPv4 CIDR
+					Type: "string",
+					Pattern: `^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4]` +
+						`[0-9]|[01]?[0-9][0-9]?)\/([0-9]|[1-2][0-9]|3[0-2])$`,
+				},
+				//{
+				//	// IPv6 CIDR
```

Comment:
Why is this commented out?

Reply:
We don't have tests for IPv6 so I just add the comment to enable it once we have tests for IPv6